### DISCUSSION TITLE
Does Jsforce Support sObject Tree Collection?

### DISCUSSION THREAD

**Discussion by ChandanShankar** [jsforce/jsforce#1468] [Ideas]
I'm trying to achieve to create a Collection of record trees, 

- Salesforce as API called sObjectTrees API,  
- [Click here](https://developer.salesforce.com/docs/atlas.en-us.api_rest.meta/api_rest/requests_composite_sobject_tree.htm) for more details.

Json Sample:
{
"records" :[{
    "attributes" : {"type" : "Account", "referenceId" : "ref1"},
    "name" : "SampleAccount",
    "phone" : "1234567890",
    "website" : "www.salesforce.com",
    "numberOfEmployees" : "100",
    "industry" : "Banking",
    "Contacts" : {
      "records" : [{
         "attributes" : {"type" : "Contact", "referenceId" : "ref2"},
         "lastname" : "Smith",
         "title" : "President",
         "email" : "sample@salesforce.com"
         },{         
         "attributes" : {"type" : "Contact", "referenceId" : "ref3"},
         "lastname" : "Evans",
         "title" : "Vice President",
         "email" : "sample@salesforce.com"
         }]
      }
    },{
    "attributes" : {"type" : "Account", "referenceId" : "ref4"},
    "name" : "SampleAccount2",
    "phone" : "1234567890",
    "website" : "www.salesforce2.com",
    "numberOfEmployees" : "100",
    "industry" : "Banking"
     }]
}

**Comment by paustint** [comment]
@ChandanShankar - Yes, any arbitrary api is supported if you use `conn.request` - I have reviewed the code, tried out various scenarios and documented my findings on this comment. Tried to get docs merged over a year ago, but nothing happened 🤷 .

https://github.com/jsforce/jsforce/issues/934#issuecomment-546714813

**Comment by cristiand391** [comment]
moved to GH discussios/ideas

We use this API in the Salesforce CLI for the `tree` commands by calling `conn.request` as mentioned above:
https://github.com/salesforcecli/plugin-data/blob/main/src/api/data/tree/importApi.ts

we could definitely move that implementation to jsforce.
